src/amber/router/pipe/reload.cr

```diff
@@ -0,0 +1,27 @@
+require "../../support/client_reload"
+
+module Amber
+  module Pipe
+    # Reload clients browsers using `ClientReload`.
+    #
+    # NOTE: Amber::Pipe::Reload is intended for use in a development environment.
+    # ```
+    # pipeline :web do
+    #   plug Amber::Pipe::Reload.new
+    # end
+    # ```
+    class Reload < Base
+      def initialize
+        Support::ClientReload.new
+        super
+      end
+
+      def call(context : HTTP::Server::Context)
+        if context.format == "html"
```

Comment:
BTW, @eliasjpr I found another issue here 😅 context.format always return "html", so the following code, gets injected too:

```crystal
class HomeController < ApplicationController
  def index
    context.response.headers["context-header"] = "application/json"
    {foo: "bar"}.to_json
  end
end
```

How can I fix this?

Reply:
The  `Accept` request header must be set in order for it to return the correct format otherwise it will return `html` by default.

Reply:
Currently working on a fix for this